srcpkgs/fwupd/template

```diff
@@ -1,23 +1,21 @@
 # Template file for 'fwupd'
 pkgname=fwupd
-version=1.5.8
+version=1.6.3
 revision=1
 build_style=meson
 build_helper="gir"
 # tests require unpackaged umockdev
 configure_args="-Dsupported_build=true
- -Dconsolekit=false -Dgtkdoc=true -Dintrospection=true
+ -Dconsolekit=false -Dintrospection=true
  -Dsystemd=false -Dplugin_altos=false -Dtests=false
- -Dpkcs7=false -Db_lto=false -Defi-includedir=${XBPS_CROSS_BASE}/usr/include/efi
- -Defi-ldsdir=${XBPS_CROSS_BASE}/usr/lib -Defi-ld=${XBPS_CROSS_BASE}/bin/ld
- -Defi-objcopy=${XBPS_CROSS_BASE}/bin/objcopy
- -Defi-libdir=${XBPS_CROSS_BASE}/usr/lib -Delogind=true"
-hostmakedepends="dejavu-fonts-ttf gnutls-tools help2man pkg-config gcab
+ -Db_lto=false -Delogind=true"
+# -Dgtkdoc=true -Dpkcs7=false
+hostmakedepends="dejavu-fonts-ttf gnutls-tools pkg-config gcab gi-docgen json-glib-devel
  vala glib-devel polkit gettext gtk-doc pango python3-gobject python3-Pillow"
 makedepends="libxmlb-devel cairo-devel colord-devel libarchive-devel
  gnutls-devel gpgme-devel json-glib-devel libgusb-devel polkit-devel
  sqlite-devel libsoup-devel gcab-devel pango-devel elogind-devel
- tpm2-tss-devel libjcat-devel libcurl-devel"
+ tpm2-tss-devel libjcat-devel libcurl-devel fwupd-efi gi-docgen valgrind-devel"
```

Comment:
* gi-docgen should be only in hostmakedepends
* it seems to build a GTK GUI, but we don't have gtk, python3-gobject and possibly other runtime dependencies listed

Reply:
Maybe we should consider splitting the GUI into its own subpackage? `fwupd` is very useful for headless usage.

Reply:
I'm not sure. The only place I see gtk in meson-options is
`option('docs', type : 'combo', choices : ['none', 'gtkdoc', 'docgen'], value : 'docgen', description : 'developer documentation type')`
so I think they just changed the way they deal with docs.
Do you want to use the value `gtkdoc` instead?
This should explain the comment below.

Reply:
Btw, if I remove it from makedepends, certain archs do fail with
```
Run-time dependency gi-docgen found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency gi-docgen
meson.build:556:2: ERROR: Automatic wrap-based subproject downloading is disabled
```

Reply:
My first suggestion is to have gi-docgen **only** in hostmakedepends, we don't need it in makedepends. I think we should always go with gi-docgen if possible, because it's more modern, some projects drop gtkdoc and it produces prettier documentation.

The second comment is unrelated to documentation and also totally wrong (my bad). I thought they ship GTK GUI, but they just use python3-gobject. So ignore the second part and @ericonr's comment

Reply:
it seems like they don't like host gi-docgen in cross compilation, I will look into it in a second

Reply:
With `gi-docgen` in both makedepends _and_ hostmakedepends, all builds are fine on ci.
If I put it only in hostmakedepends, then armv7l and aarch fail, with the error above.

Reply:
exactly

Reply:
Something wrong with how they look for `gi-docgen`, then. Should be fixable, but I'd be okay with having it in both vars.

Reply:
It's very simple to fix, could you add this PR as a patch? https://github.com/fwupd/fwupd/pull/3664

Reply:
Sure.